src/webportal/config/webpack.common.js

```diff
@@ -17,67 +17,68 @@
 
 
 // module dependencies
-const fs = require('fs');
 const path = require('path');
 const webpack = require('webpack');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
 const CopyWebpackPlugin = require('copy-webpack-plugin');
-const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 const MonacoWebpackPlugin = require('monaco-editor-webpack-plugin');
-const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+const TerserPlugin = require('terser-webpack-plugin');
 const markedConfig = require('./marked.config');
 const helpers = require('./helpers');
 
 const title = 'Platform for AI';
 const version = require('../package.json').version;
 const FABRIC_DIR = path.resolve(__dirname, '../src/app/job/job-view/fabric');
 
+const devMode = process.env.NODE_ENV !== 'production';
 
-const htmlMinifierOptions = {
-  collapseWhitespace: true,
-  html5: true,
-  minifyCSS: true,
-  minifyJS: true,
-  removeComments: true,
-  removeEmptyAttributes: true,
-  removeTagWhitespace: true
-};
+function generateHtml(opt) {
+  return new HtmlWebpackPlugin(Object.assign({
+    title: title,
+    version: version,
+    template: './src/app/layout/layout.component.ejs',
+    minify: {
```

Comment:
Prevent minification during development may increase the build speed.

Reply:
JS minify is disabled in dev mode. And I think html minify is fast